modules/cloudformation__download_data/main.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+from botocore.exceptions import ClientError
+import json, os, datetime
+from json import JSONEncoder
+from core.secretfinder.utils import regex_checker, Color
+module_info = {'name':'cloudformation__download_data', 
+ 'author':'David Yesland', 
+ 'category':'ENUM', 
+ 'one_liner':'Downloads all templates, parameters, and exports from CloudFormation Stacks.', 
+ 'description':'Downloads all templates, parameters, and exports from CloudFormation Stacks. Looks for secrets in all and saves data to files.', 
+ 'services':[
+  'cloudformation'], 
+ 'prerequisite_modules':[],  'external_dependencies':[],  'arguments_to_autocomplete':[]}
+parser = argparse.ArgumentParser(add_help=False, description=(module_info['description']))
+
+parser.add_argument('--regions', required=False, default=None, help='One or more (comma separated) AWS regions in the format us-east-1. Defaults to all regions.')
+
+def main(args, pacu_main):
+    session = pacu_main.get_active_session()
+    print = pacu_main.print
+    get_regions = pacu_main.get_regions
+    regions = get_regions('cloudformation')
+
+    def find_secrets(string):
+        detections = regex_checker(string)
+        [Color.print(Color.GREEN, '\tDetected {}: {}'.format(itemkey, detections[itemkey])) for itemkey in detections]
+
+    def outfile(subdir, filename):
+        base_path = 'sessions/{}/downloads/{}/{}/'.format(session.name, module_info['name'], subdir)
+        if not os.path.exists(base_path):
+            os.makedirs(base_path)
+        return open(base_path + filename, 'a+')
+
+    class DateTimeEncoder(JSONEncoder):
+
+        def default(self, obj):
+            if isinstance(obj, (datetime.date, datetime.datetime)):
+                return obj.isoformat()
+
+    stacks = []
+    found_regions = []
+    for region in regions:
+        client = pacu_main.get_boto3_client('cloudformation', region)
+        print('Looking for CloudFormation Stacks in region {}...'.format(region))
+        stacks_data = client.describe_stacks()
+        stacks += stacks_data['Stacks']
```

Comment:
Looks like we end up getting duplicate reports in the output because this isn't reset for each region.

The for loop below ends up getting executed for each stack in all the regions run so far.